### ISSUE TITLE
Add a way of uploading demos

### ISSUE THREAD

**Issue by splewis** [splewis/get5-web#4]
Either uploaded directly to the web server by the game server plugin, or uploaded elsewhere (e.g. S3) and accessible via a url that is displayed on the match page.

**Comment by BloodyIron** [comment]
Any news on this? I would LOVE this feature!

**Comment by mloveless1227** [comment]
I know this isn't the exact answer people are looking for, but as a temporary solution you could use Thrawn's EasyFTP and Demo upload plugins for Sourcemod. I use them alongside this plugin and it autorecords & uploads to my website after  each match no problem. 

http://www.sourcemod.net/plugins.php?cat=0&mod=-1&title=&author=thrawn&description=&search=1

In the above link, use tEasyFTP and AutoDemoUpload, I think there are a couple other requirements with each but you obviously have to read into each plugin. My apologies if you don't want this here, I just came across this issue and wanted to offer something that works for me until the feature is potentially added.

**Comment by MikkelDK** [comment]
@mloveless1227 not all, know how to setup, thats why people is just waiting. ;) (meself i still dont know. since i don't host or playing anymore)

**Comment by BloodyIron** [comment]
When you're running a LAN with lots of CSGO servers, that makes it very hard to keep track of. I for one want get5 to handle it so we can more easily administrate and keep records and demos organised.

**Comment by BloodyIron** [comment]
FYI, when I submitted ( https://github.com/splewis/get5/issues/354 ) I really should have pointed it to here.

Namely because I am very interested in the CSGO match demos being uploaded to the get5-web server and tied to each match. So players and admins can download them in the central webGUI and know which demo is for which match, etc.

Can we have this topic revisited please? I think a lot of this work really only needs to be done once, and since I believe the get5 servers already record demos by default, not a lot of connecting work needs to be done.

I'm not a developer myself, I'm a TO, and this would help me a tonne, and I'm sure plenty of other events would love this too.

Please? :)

**Comment by PhlexPlexico** [comment]
Splewis has already set several times in these tickets that this is a proof of concept and not maintained. If you would like to see this happen, you will need to DIY.

**Comment by BloodyIron** [comment]
So we're not allowed to make feature requests at all on here? Why even have a "issue" section? -.-

**Comment by splewis** [comment]
You can make whatever requests or bug reports you want. They're just extremely unlikely to be acted upon.

**Comment by PhlexPlexico** [comment]
Fair point, and apologies if I came off as rude, didn't mean to. And I don't want to advertise the fork I created, but there are solutions that are (attempting) to be put in place. 

And you do have a good point about the issues being a place for ideas, so I do apologise for that.

As for PRs into this repo, are they unlikely to be acted upon as well?

**Comment by BloodyIron** [comment]
I just want to clarify that I'm not trying to come in here and bark orders or anything like that. I know this is volunteer development work going on.

I'm just hoping I can make some convincing arguments that may be successful enough to get features I care about added. With any luck "soon-ish" (whatever that means).

As to what happens in reality, I know that's beyond my control. But in-spite of that, I hope I am successful in making compelling points, because I sure would love to get demos central on get5-web ;)

@PhlexPlexico I accept and appreciate your apology.

@splewis any chance this could be added to whatever "queue" may exist? :)

**Comment by topjor** [comment]
I would like to let you know I am working on a demo upload feature for the apistats plugin
I am currently building my own web panel for get5 using the current apistats plugin so the panel part I will probably not be doing, also I am not that known with python. (I am building this panel for a lan party organization as a side project (publicly open sourcing is on the list))

my repository of my forked apistats plugin: https://gitlab.j0dev.nl/lan-projects/get5/web5-apistats
(the demo upload system if currently in a different branch: 1-demo-upload-system)

I will look into upstreaming this when it is finished, but I was also thinking about putting the apistats plugin into a separate repository, but I dont know what you guys think

**Comment by splewis** [comment]
>  will look into upstreaming this when it is finished, but I was also thinking about putting the apistats plugin into a separate repository, but I don't know what you guys think

I would suggest making it entirely separate, and not aiming for compatibility with the existing apistats plugin or web panel. Those are really intended more as examples for doing exactly what you're doing here.  

I hope your project goes well!

**Comment by nishuo** [comment]
Features worth looking forward to! It seems that he is almost done!
